### ISSUE TITLE
Medusa monorepo with turborepo doesn't start because of `@medusajs/admin`

### ISSUE THREAD

**Issue by abathellier** [medusajs/medusa#5120]
### Describe the bug

I'm encountering a bug when trying to integrate MedusaJS into a monorepo with [TurboRepo](https://turbo.build/repo). When attempting to run MedusaJS in this context, I'm receiving the following error:

```
[@medusajs/admin] resolve 'swc-loader' in '/Users/user/Documents/web/medusa-turborepo/apps/medusa'
```

This issue seems to be specific to `@medusajs/admin` v7, as it was working with the v6 in a similar setup.

You can see a reproduction there : https://github.com/durdenx/medusa-turborepo. 

- `main` branch for the non-working solution
- `working-with-medusa-admin-v6` branch for the working solution

### System information

- Medusa version (including plugins):   
  - "@medusajs/medusa": "^1.16.0"
  - "@medusajs/admin": "^7.1.0",
- Node.js version: v18.17.1
- Database: Postgres 15
- Operating system: macOS

### Steps to reproduce the behavior

1. Clone the repository: `git clone https://github.com/durdenx/medusa-turborepo`
2. Navigate to the project directory: `cd medusa-turborepo`
3. Setup .env (rename .env.template to .env)
4. Install database: `docker-compose up -d`
5. Install dependencies: `pnpm install`
6. Try to run : `pnpm dev`

### Expected behavior

I expected MedusaJS to run successfully within the TurboRepo monorepo environment.

### Additional context

Related to #4532 and #5544

Any help or suggestions to resolve this issue would be greatly appreciated.

Thank you in advance for your assistance!

**Comment by abdokouta** [comment]
Same issue for Nx monorepos

**Comment by aliameur** [comment]
Any update on this?

**Comment by olivermrbl** [comment]
@aliameur, are you experiencing the issue using turborepo?

**Comment by olivermrbl** [comment]
@durdenx, I am unable to reproduce your issue following the steps (with out docker, though). Could I get you to try again, so I can be sure the issue still exist for turborepo setups?

**Comment by abathellier** [comment]
@olivermrbl  Yes, I temporarily set `shamefully-hoist = true` in the `.npmrc` file as a workaround. Just remove this setting to get back to the initial issue.

**Comment by olivermrbl** [comment]
@durdenx, any chance I could get you to update the repo (or a branch in the repo) to the reproducible state? I would like to be sure we have 1:1 setups.

**Comment by abathellier** [comment]
@olivermrbl, I've updated the repository. Please make sure to run `pnpm clean` and `pnpm install` from the root to see the error.

**Comment by aliameur** [comment]
@olivermrbl , no I am using a nx monorepo. It seems that `@meduajs/admin` needs its depedencies in the same directory, and when you hack around it with nx, you start getting getting a load of dependency issues from the .cache folder.

**Comment by olivermrbl** [comment]
Can I get you to try the following snapshots:
```
@medusajs/medusa@1.19.1-snapshot-20240108132634
@medusajs/medusa-cli@1.3.23-snapshot-20240108132634
```

Any or all of you.

**Comment by abathellier** [comment]
@olivermrbl It's not working. After updating, I still encounter the same error: `Module not found: Error: Can't resolve 'swc-loader'`

**Comment by olivermrbl** [comment]
Sorry, I should have added that my attempt will not solve the issue with the `swc-loader`, only the issues related to babel and medusa executables. I realise this is what was initially reported – I apologise for the confusion my comment may have caused. 

If you install the `swc-loader` package (and the other failing one) separately in your project, as mentioned [here](https://github.com/medusajs/medusa/issues/4532#issuecomment-1700908959), you should be able to resolve your issue. 

This is not ideal. However, the issue will be permanently eliminated as we update our admin at the end of this month.

**Comment by abathellier** [comment]
Even when I install `swc-loader`, I then need to install all the admin dependencies (react, radix, etc.), and I encounter the following error: `[@medusajs/admin] Could not find the admin UI build files. Please run "medusa-admin build" or enable "autoRebuild" in the plugin options to build the admin UI.`

For now, adding `shamefully-hoist = true` in `.npmrc` is working well for me. I will wait for you to update Medusa Admin at the end of the month.

**Comment by facundocapua** [comment]
For is happening too, the workaround suggested by @durdenx works for me

**Comment by buffcode** [comment]
Any updates on this? In which version is / was this expected to be fixed?

We have a monorepo and `shamefully-hoist = true` only works globally within the repo and cannot be restricted to the medusa part only.
